chore(#8350): move cht-core/tests /AUTOMATE_TEST_GUIDE.md file (#1461)
medic/cht-docs: content/en/contribute/code/core/automated-tests.md

    e2e-tests o--Pouch/HTTPS--o cht-e2e
```

> ### Tips to write automated e2e tests

Reviewer: ```suggestion
### How to write automated e2e tests
```

> ### Tips to write automated e2e tests

Reviewer: ```suggestion
### How to write automated e2e tests
```

> Please read the [style guide for automated tests]({{< relref "contribute/code/core/style-guide-automated-e2e-tests.md" >}}) which provides editorial guidelines for anyone creating new automated test cases for CHT-Core.

Author: Same here, I am not sure if I linked the file correctly

Reviewer: Linked correctly 👌🏽.

> Please read the [style guide for automated tests]({{< relref "contribute/code/core/style-guide-automated-e2e-tests.md" >}}) which provides editorial guidelines for anyone creating new automated test cases for CHT-Core.

Reviewer: ```suggestion
Read the [style guide for automated tests]({{< relref "contribute/code/core/style-guide-automated-e2e-tests.md" >}}) to get guidelines on creating new automated test cases for CHT-Core.
```

> Please read the [style guide for automated tests]({{< relref "contribute/code/core/style-guide-automated-e2e-tests.md" >}}) which provides editorial guidelines for anyone creating new automated test cases for CHT-Core.

Reviewer: ```suggestion
Read the [style guide for automated tests]({{< relref "contribute/code/core/style-guide-automated-e2e-tests.md" >}}) for guidelines on how to create new automated test cases for CHT-Core.
```

#### Other logs and screenshots

GitHub actions will artifact all files in tests/logs. This is the directory any logs, results, images, etc... should save to if you want to review them if a build fails.

##### View the CI report
There are logs and screenshots stored in the allure reports when a job failed on the CI. To access to those logs follow these steps:
- Download the CI run artifact zip file located in the failed build's `Archive Results` section.

>   {{< figure src="archiveResultsSection.png" link="archiveResultsSection.png" class=" center col-4 col-lg-4" >}}

Author: @esthermmoturi I am not sure if did this correctly, could you please chek?

Reviewer: Yes, the image wasn't renderring because the folder with the images is `automated-test` instead of `automated-tests`. I'll make the change.

>   {{< figure src="archiveResultsSection.png" link="archiveResultsSection.png" class=" center col-4 col-lg-4" >}}

Reviewer: ```suggestion
  {{< figure src="archiveResultsSection.png" link="archiveResultsSection.png" class=" center col-12 col-lg-12" >}}
```

Try the following:
- Manually downloaded the images. To download images manually, you can use either docker-compose or docker:
  - With docker, you'd do a docker pull  <image tag> for every image you want to download.
  - With docker-compose, you'd save all docker-compose files in a folder, do a docker-compose pull, and point to your files as a source. Read more on [docker compose pull](https://docs.docker.com/engine/reference/commandline/compose_pull/)

### Test Architecture

> Our GitHub actions spin up an ubuntu-22.04 machine. Installs software and then launches Couchdb and Horticulturalist in a docker container. This is needed to run our applications in the specific node versions we support while allowing our test code to run in versions of node it they support. This creates a paradigm to keep in mind when writing tests. Tests run on the ubuntu machine. Any test code that starts a server or runs an executable is running outside of the horti container. The ports are exposed for all our services and horti has access to the cht-core root via a volume. Horti can also talk to the host by getting the gateway of the docker network. 

Reviewer: ```suggestion
Our GitHub actions spin up an ubuntu-22.04 machine. They install software and then launch couchdb and horticulturalist in a docker container. This is needed to run our applications in the specific node versions we support, while allowing our test code to run in versions of node it supports. This creates a paradigm to keep in mind when writing tests. Tests run on the ubuntu machine. 
Any test code that starts a server or runs an executable is running outside of the horti container. The ports are exposed for all our services and horti has access to the cht-core root via a volume. Horti can also talk to the host by getting the gateway of the docker network. 
```

- Manually downloaded the images. To download images manually, you can use either docker-compose or docker:
  - With docker, you'd do a docker pull  <image tag> for every image you want to download.

>   - With docker-compose, you'd save all docker-compose files in a folder, do a docker-compose pull, and point to your files as a source. [Compose pull documentation](https://docs.docker.com/engine/reference/commandline/compose_pull/)

Reviewer: ```suggestion
  - With docker-compose, you'd save all docker-compose files in a folder, do a docker-compose pull, and point to your files as a source. Read more on [docker compose pull](https://docs.docker.com/engine/reference/commandline/compose_pull/)
```
